internal/gatewayapi/helpers.go

```diff
@@ -314,6 +314,37 @@ func computeHosts(routeHostnames []string, listenerHostname *gwapiv1.Hostname) [
 	return hostnames
 }
 
+// removeIsolatedListeners returns a list of the hostnames between the route
```

Comment:
curious why this is needed ? is the issue that `computeHosts(GetHostnames)` is returning duplicates ?
can't we prune that o/p instead ?

Reply:
`computeHosts` returns a list of hostnames based on the intersection between the HTTPRoute's hostnames and a single parentRef's listener each time. 
However, to make this intersection correct with hostname isolation, we need to take into account the entire list of parentRef's listeners.

We can make the change inside `computeHosts` by passing the entire listener each time (instead of passing only `listener.Hostname`. I just thought it would be clearer this way.

@arkodg Please let me know if you want the change to be in `computeHosts` and I'll update the PR.

Reply:
ah thanks for clarifying the current limitation
would prefer if we redefined `computeHosts` to return  - `intersecting listener hostnames and route hostnames that dont intersect with other listener hostnames`

Reply:
100%
I'll make the change